extra/fuegogin/context.go

```diff
@@ -99,12 +103,27 @@ func (c ginContext[B]) SetStatus(code int) {
 }
 
 func (c ginContext[B]) Serialize(data any) error {
-	c.ginCtx.JSON(http.StatusOK, data)
+	if c.DefaultStatusCode == 0 {
+		c.DefaultStatusCode = http.StatusOK
+	}
+	status := c.ginCtx.Writer.Status()
+	if status == 0 {
+		status = c.DefaultStatusCode
+	}
+	if status == 0 {
+		status = 200
+	}
+	c.ginCtx.JSON(status, data)
 	return nil
```

Comment:
```suggestion
	status := c.ginCtx.Writer.Status()
	if status == 0 {
		status = c.DefaultStatusCode
	}
	if status == 0 {
		status = http.StatusOK
	}
	c.ginCtx.JSON(status, data)
 	return nil
```

Or


```suggestion
	if c.DefaultStatusCode == 0 {
		c.DefaultStatusCode = http.StatusOK
	}
	status := c.ginCtx.Writer.Status()
	if status == 0 {
		status = c.DefaultStatusCode
	}
	c.ginCtx.JSON(status, data)
 	return nil
```

Because otherwise you are doing twice